File: frontend/src/api/coffeeBean.ts
import request from './request'
import { PaginatedResponse } from './coffeeLog'

export interface CoffeeBean {
  id: number
  user_id: number
  name: string
  origin: string
  processing_method: string
  roast_level: string
  roaster: string
  image_url: string
  usage_count: number
  created_at: string
  updated_at: string
}

export interface CreateCoffeeBeanParams {
  name: string
  origin?: string
  processing_method?: string
  roast_level?: string
  roaster?: string
  image_url?: string
}

export interface UpdateCoffeeBeanParams {
  name?: string
  origin?: string
  processing_method?: string
  roast_level?: string
  roaster?: string
  image_url?: string
}

export async function getCoffeeBeans(params?: {
  page?: number
  page_size?: number
  search?: string
}): Promise<PaginatedResponse<CoffeeBean>> {
  return request.get('/coffee-beans', { params }) as unknown as Promise<PaginatedResponse<CoffeeBean>>
}

export async function getCoffeeBeanById(id: number): Promise<CoffeeBean> {
  return request.get(`/coffee-beans/${id}`) as unknown as Promise<CoffeeBean>
}

export async function createCoffeeBean(params: CreateCoffeeBeanParams): Promise<CoffeeBean> {
  return request.post('/coffee-beans', params) as unknown as Promise<CoffeeBean>
}

export async function updateCoffeeBean(id: number, params: UpdateCoffeeBeanParams): Promise<CoffeeBean> {
  return request.put(`/coffee-beans/${id}`, params) as unknown as Promise<CoffeeBean>
}

export async function deleteCoffeeBean(id: number) {
  return request.delete(`/coffee-beans/${id}`)
}

export async function getBeanList(): Promise<CoffeeBean[]> {
  const res = await request.get('/coffee-beans/list') as any
  return res.beans || []
}

File: backend/internal/model/coffee_bean.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type CoffeeBean struct {
	ID              uint           `json:"id" gorm:"primaryKey"`
	UserID          uint           `json:"user_id" gorm:"not null;index"`
	Name            string         `json:"name" gorm:"size:255;not null"`
	Origin          string         `json:"origin" gorm:"size:255"`
	ProcessingMethod string        `json:"processing_method" gorm:"size:100"`
	RoastLevel      string         `json:"roast_level" gorm:"size:50"`
	Roaster         string         `json:"roaster" gorm:"size:255"`
	ImageURL        string         `json:"image_url" gorm:"size:500"`
	UsageCount      int            `json:"usage_count" gorm:"default:0"`
	CreatedAt       time.Time      `json:"created_at"`
	UpdatedAt       time.Time      `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `json:"deleted_at" gorm:"index"`

	User User `json:"-" gorm:"foreignKey:UserID"`
}
